Dockerfile generator: scope cargo build to the dashboard crate (-p) to avoid building wasm-incompatible workspace members

## Summary

The auto-generated `Dockerfile`'s wasm stage runs

```dockerfile
RUN cargo build --release --target wasm32-unknown-unknown
```

without a `-p <crate>` filter, so cargo tries to compile **every** workspace member for `wasm32-unknown-unknown`. In the Reinhardt Cloud workspace 7 of those members (operator, agent, CLI, k8s, grpc, core, telemetry) depend on `tokio = { features = ["full"] }`, which transitively pulls `mio`'s `net` feature. `mio`'s net feature does not compile for `wasm32`, so the build dies with **48 errors in `mio`**.

This is a follow-up to #477 (which only added `protoc` install but left the `cargo build` command itself unchanged).

## Reproduction

1. Apply the protoc fix from #477 / PR #482 (already merged) to `dashboard/Dockerfile`.
2. Run `docker build -f dashboard/Dockerfile -t e2e .` from the workspace root.
3. The `[wasm 7/8]` stage fails with:
   ```
   error: This wasm target is unsupported by mio. If using Tokio, disable the net feature.
   error: could not compile `mio` (lib) due to 48 previous errors
   ```
4. Validated against worktree `chore/reinhardt-web-main-refactor` at `28437e1c` (PR #469), with the regenerated Dockerfile (`28KB` build log: 19.20s into the wasm stage).

## Root Cause

`crates/reinhardt-cloud-cli/src/dockerfile_generator/stages.rs` emits

```rust
RUN cargo build --release --target wasm32-unknown-unknown
```

without `-p <package>`. Cargo with `--target` builds every member that does not opt out of the target, so the wasm-incompatible operator/agent/CLI/grpc crates are pulled into the build graph even though only `dashboard`'s `cdylib` artifact is needed.

`dashboard/Cargo.toml` already correctly gates its server-only deps:

```toml
[target.'cfg(not(target_arch = "wasm32"))'.dependencies]
reinhardt = { workspace = true }
# ... server-only crates ...
tokio = { workspace = true }

[target.'cfg(target_arch = "wasm32")'.dependencies]
reinhardt = { workspace = true, features = ["pages", "admin"] }
wasm-bindgen = "0.2"
# ... wasm-only crates, no tokio ...
```

So compiling **only** `-p reinhardt-cloud-dashboard --target wasm32-unknown-unknown` would be wasm-clean.

## Suggested Fix

Update `dockerfile_generator/stages.rs` to emit the package name from `reinhardt-cloud.toml` (or detect from `Cargo.toml` `[package].name`):

```dockerfile
# wasm stage
RUN cargo build --release --target wasm32-unknown-unknown -p <package-name>
RUN wasm-bindgen ... target/wasm32-unknown-unknown/release/<package_underscored>.wasm ...

# builder stage (also benefits)
RUN cargo build --release -p <package-name>
```

Symmetric arguments apply to the `builder` stage: today it builds the entire workspace's `--release` set, which includes the operator/agent/CLI binaries — wasted work in a dashboard image (~3-5 minutes saved).

## Workaround

Hand-edit `dashboard/Dockerfile`:

```dockerfile
RUN cargo build --release --target wasm32-unknown-unknown -p reinhardt-cloud-dashboard
```

But this is overwritten the next time `reinhardt-cloud init` regenerates the file.

## Related

- Closes #477 covered protoc install only; this is the orthogonal "scope the build to the package" fix.
- The `libprotobuf-dev` install for prost-build's well-known type headers is **also still missing** (see #477 follow-up comment) — surfacing a separate issue for that as well.

## Acceptance Criteria

- [ ] `dockerfile_generator` reads the package name from `reinhardt-cloud.toml` (or `Cargo.toml`) and passes `-p <name>` to both `cargo build` invocations.
- [ ] `wasm-bindgen` invocation uses the same crate name (snake_cased) instead of hardcoded.
- [ ] Snapshot tests in `dockerfile_generator/snapshots/*.dockerfile` updated.
- [ ] Verified by rebuilding `dashboard/Dockerfile` end-to-end.

## Verification Evidence

### Repro Build Failure (auto-generated Dockerfile, post-PR-#482)

```
#19 [wasm 7/8] RUN cargo build --release --target wasm32-unknown-unknown
...
#19 18.87 error[E0433]: failed to resolve: could not find `Events` in `sys`
#19 18.87   --> /usr/local/cargo/registry/src/index.crates.io-1949cf8c6b5b557f/mio-1.2.0/src/event/events.rs:94:25
#19 18.87 error[E0425]: cannot find function `set_reuseaddr` in this scope
#19 19.20 error: could not compile `mio` (lib) due to 48 previous errors
ERROR: failed to build: failed to solve: process "/bin/sh -c cargo build --release --target wasm32-unknown-unknown" did not complete successfully: exit code: 101
```

The 48 errors are all `mio` complaining about its `net` feature on `wasm32`, pulled in by tokio with `features = ["full"]` which 7 workspace members declare:

```
crates/reinhardt-cloud-agent/Cargo.toml
crates/reinhardt-cloud-grpc/Cargo.toml
crates/reinhardt-cloud-core/Cargo.toml
crates/reinhardt-cloud-cli/Cargo.toml
crates/reinhardt-cloud-k8s/Cargo.toml
crates/reinhardt-cloud-telemetry/Cargo.toml
crates/reinhardt-cloud-operator/Cargo.toml
```

### Workaround Verification

Patched `Dockerfile.custom`:

```dockerfile
RUN cargo chef cook --release --recipe-path recipe.json -p reinhardt-cloud-dashboard
RUN cargo build --release -p reinhardt-cloud-dashboard --bin reinhardt-cloud-dashboard
RUN cargo build --release --target wasm32-unknown-unknown -p reinhardt-cloud-dashboard
```

E2E build with this patch is in progress; I'll post the result once it finishes.

### Fix Location

`crates/reinhardt-cloud-cli/src/dockerfile_generator/stages.rs` — both the builder stage's `cargo build --release` (line emitting after the `COPY . .`) and the wasm stage's `cargo build --release --target wasm32-unknown-unknown` need to take a `--package <name>` argument, with `<name>` resolved from the same `[application]` block in `reinhardt-cloud.toml` that already drives `wasm-bindgen`'s `<crate>.wasm` filename.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/crates/reinhardt-cloud-cli/src/dockerfile_generator/snapshots/api_full.dockerfile
+++ b/crates/reinhardt-cloud-cli/src/dockerfile_generator/snapshots/api_full.dockerfile
@@ -17,7 +17,7 @@ WORKDIR /app
 COPY --from=chef /app/recipe.json recipe.json
 RUN cargo chef cook --release --features graphql --recipe-path recipe.json
 COPY . .
-RUN cargo build --release --features graphql
+RUN cargo build --release -p my-app --features graphql
 
 FROM debian:bookworm-slim AS runtime
 RUN apt-get update && \

--- a/crates/reinhardt-cloud-cli/src/dockerfile_generator/snapshots/api_graphql.dockerfile
+++ b/crates/reinhardt-cloud-cli/src/dockerfile_generator/snapshots/api_graphql.dockerfile
@@ -11,7 +11,7 @@ WORKDIR /app
 COPY --from=chef /app/recipe.json recipe.json
 RUN cargo chef cook --release --features graphql --recipe-path recipe.json
 COPY . .
-RUN cargo build --release --features graphql
+RUN cargo build --release -p my-app --features graphql
 
 FROM debian:bookworm-slim AS runtime
 RUN apt-get update && \

--- a/crates/reinhardt-cloud-cli/src/dockerfile_generator/snapshots/api_grpc.dockerfile
+++ b/crates/reinhardt-cloud-cli/src/dockerfile_generator/snapshots/api_grpc.dockerfile
@@ -17,7 +17,7 @@ WORKDIR /app
 COPY --from=chef /app/recipe.json recipe.json
 RUN cargo chef cook --release --recipe-path recipe.json
 COPY . .
-RUN cargo build --release
+RUN cargo build --release -p my-app
 
 FROM debian:bookworm-slim AS runtime
 RUN apt-get update && \

--- a/crates/reinhardt-cloud-cli/src/dockerfile_generator/snapshots/api_minimal.dockerfile
+++ b/crates/reinhardt-cloud-cli/src/dockerfile_generator/snapshots/api_minimal.dockerfile
@@ -11,7 +11,7 @@ WORKDIR /app
 COPY --from=chef /app/recipe.json recipe.json
 RUN cargo chef cook --release --recipe-path recipe.json
 COPY . .
-RUN cargo build --release
+RUN cargo build --release -p my-app
 
 FROM debian:bookworm-slim AS runtime
 RUN apt-get update && \

--- a/crates/reinhardt-cloud-cli/src/dockerfile_generator/snapshots/api_mysql.dockerfile
+++ b/crates/reinhardt-cloud-cli/src/dockerfile_generator/snapshots/api_mysql.dockerfile
@@ -11,7 +11,7 @@ WORKDIR /app
 COPY --from=chef /app/recipe.json recipe.json
 RUN cargo chef cook --release --recipe-path recipe.json
 COPY . .
-RUN cargo build --release
+RUN cargo build --release -p my-app
 
 FROM debian:bookworm-slim AS runtime
 RUN apt-get update && \

--- a/crates/reinhardt-cloud-cli/src/dockerfile_generator/snapshots/api_postgres.dockerfile
+++ b/crates/reinhardt-cloud-cli/src/dockerfile_generator/snapshots/api_postgres.dockerfile
@@ -11,7 +11,7 @@ WORKDIR /app
 COPY --from=chef /app/recipe.json recipe.json
 RUN cargo chef cook --release --recipe-path recipe.json
 COPY . .
-RUN cargo build --release
+RUN cargo build --release -p my-app
 
 FROM debian:bookworm-slim AS runtime
 RUN apt-get update && \

--- a/crates/reinhardt-cloud-cli/src/dockerfile_generator/snapshots/api_sqlite.dockerfile
+++ b/crates/reinhardt-cloud-cli/src/dockerfile_generator/snapshots/api_sqlite.dockerfile
@@ -11,7 +11,7 @@ WORKDIR /app
 COPY --from=chef /app/recipe.json recipe.json
 RUN cargo chef cook --release --recipe-path recipe.json
 COPY . .
-RUN cargo build --release
+RUN cargo build --release -p my-app
 
 FROM debian:bookworm-slim AS runtime
 RUN apt-get update && \

--- a/crates/reinhardt-cloud-cli/src/dockerfile_generator/snapshots/pages_full.dockerfile
+++ b/crates/reinhardt-cloud-cli/src/dockerfile_generator/snapshots/pages_full.dockerfile
@@ -17,7 +17,7 @@ WORKDIR /app
 COPY --from=chef /app/recipe.json recipe.json
 RUN cargo chef cook --release --features graphql --recipe-path recipe.json
 COPY . .
-RUN cargo build --release --features graphql
+RUN cargo build --release -p my-app --features graphql
 
 FROM rust:1.94.1-bookworm AS wasm
 RUN apt-get update && \
@@ -27,7 +27,7 @@ RUN rustup target add wasm32-unknown-unknown
 RUN cargo install wasm-bindgen-cli@0.2.100
 WORKDIR /app
 COPY . .
-RUN cargo build --release --target wasm32-unknown-unknown
+RUN cargo build --release --target wasm32-unknown-unknown --lib -p my-app
 RUN wasm-bindgen --out-dir /wasm-dist --target web target/wasm32-unknown-unknown/release/my_app.wasm && \
     wasm-opt -Oz -o /wasm-dist/optimized.wasm /wasm-dist/*.wasm
 

--- a/crates/reinhardt-cloud-cli/src/dockerfile_generator/snapshots/pages_minimal.dockerfile
+++ b/crates/reinhardt-cloud-cli/src/dockerfile_generator/snapshots/pages_minimal.dockerfile
@@ -11,7 +11,7 @@ WORKDIR /app
 COPY --from=chef /app/recipe.json recipe.json
 RUN cargo chef cook --release --recipe-path recipe.json
 COPY . .
-RUN cargo build --release
+RUN cargo build --release -p my-app
 
 FROM rust:1.94.1-bookworm AS wasm
 RUN apt-get update && \
@@ -21,7 +21,7 @@ RUN rustup target add wasm32-unknown-unknown
 RUN cargo install wasm-bindgen-cli@0.2.100
 WORKDIR /app
 COPY . .
-RUN cargo build --release --target wasm32-unknown-unknown
+RUN cargo build --release --target wasm32-unknown-unknown --lib -p my-app
 RUN wasm-bindgen --out-dir /wasm-dist --target web target/wasm32-unknown-unknown/release/my_app.wasm && \
     wasm-opt -Oz -o /wasm-dist/optimized.wasm /wasm-dist/*.wasm
 

--- a/crates/reinhardt-cloud-cli/src/dockerfile_generator/snapshots/pages_postgres.dockerfile
+++ b/crates/reinhardt-cloud-cli/src/dockerfile_generator/snapshots/pages_postgres.dockerfile
@@ -11,7 +11,7 @@ WORKDIR /app
 COPY --from=chef /app/recipe.json recipe.json
 RUN cargo chef cook --release --recipe-path recipe.json
 COPY . .
-RUN cargo build --release
+RUN cargo build --release -p my-app
 
 FROM rust:1.94.1-bookworm AS wasm
 RUN apt-get update && \
@@ -21,7 +21,7 @@ RUN rustup target add wasm32-unknown-unknown
 RUN cargo install wasm-bindgen-cli@0.2.100
 WORKDIR /app
 COPY . .
-RUN cargo build --release --target wasm32-unknown-unknown
+RUN cargo build --release --target wasm32-unknown-unknown --lib -p my-app
 RUN wasm-bindgen --out-dir /wasm-dist --target web target/wasm32-unknown-unknown/release/my_app.wasm && \
     wasm-opt -Oz -o /wasm-dist/optimized.wasm /wasm-dist/*.wasm
 

--- a/crates/reinhardt-cloud-cli/src/dockerfile_generator/stages.rs
+++ b/crates/reinhardt-cloud-cli/src/dockerfile_generator/stages.rs
@@ -121,8 +121,13 @@ pub(crate) fn build_builder_stage(signals: &DockerfileSignals) -> Stage {
 		src: ".".to_string(),
 		dst: ".".to_string(),
 	});
+	// Scope the build to the project crate so wasm-incompatible workspace
+	// members (operator/agent/CLI/grpc/etc.) are not pulled into the build
+	// graph. See kent8192/reinhardt-cloud#485 for the underlying mio failure
+	// that motivated scoping.
 	instructions.push(Instruction::Run(format!(
-		"cargo build --release{feature_args}"
+		"cargo build --release -p {}{feature_args}",
+		signals.app_name
 	)));
 
 	Stage {
@@ -161,7 +166,17 @@ pub(crate) fn build_wasm_stage(signals: &DockerfileSignals) -> Stage {
 				src: ".".to_string(),
 				dst: ".".to_string(),
 			},
-			Instruction::Run("cargo build --release --target wasm32-unknown-unknown".to_string()),
+			// `--lib -p {app_name}` ensures only the dashboard crate's cdylib is
+			// built. Without `-p` cargo builds every workspace member for the
+			// wasm32 target, which fails on members that depend on
+			// `tokio = { features = ["full"] }` because `mio`'s net feature is
+			// not wasm-compatible. Without `--lib`, cargo also tries to build
+			// the project's host binary (`main.rs`) for wasm32 and fails.
+			// See kent8192/reinhardt-cloud#485.
+			Instruction::Run(format!(
+				"cargo build --release --target wasm32-unknown-unknown --lib -p {}",
+				signals.app_name
+			)),
 			Instruction::RunMulti(vec![
 				format!(
 					"wasm-bindgen --out-dir /wasm-dist --target web \
@@ -608,9 +623,10 @@ mod tests {
 
 		// Assert: both cargo-chef caching and the final build must carry the
 		// graphql feature so the compiled binary includes GraphQL code.
+		// The cargo build line also carries `-p my-app` for #485 scoping.
 		assert!(
-			stage_contains_run(&stage, "cargo build --release --features graphql"),
-			"expected graphql feature in cargo build"
+			stage_contains_run(&stage, "cargo build --release -p my-app --features graphql"),
+			"expected graphql feature in cargo build with package scope"
 		);
 		assert!(
 			stage_contains_run(&stage, "cargo chef cook --release --features graphql"),
@@ -723,6 +739,46 @@ mod tests {
 		assert!(
 			stage_contains_run(&stage, "protobuf-compiler"),
 			"builder stage must install protobuf-compiler when protoc_needed is set"
+		);
+	}
+
+	// PS1 (Refs #485): builder stage scopes cargo build to the project
+	// package, preventing wasm-incompatible workspace members from being
+	// pulled into the build graph.
+	#[rstest]
+	fn builder_stage_scopes_cargo_build_to_package(minimal_signals: DockerfileSignals) {
+		// Act
+		let stage = build_builder_stage(&minimal_signals);
+
+		// Assert
+		assert!(
+			stage_contains_run(&stage, "cargo build --release -p my-app"),
+			"builder stage must pass `-p {{app_name}}` to cargo build; \
+			 see kent8192/reinhardt-cloud#485"
+		);
+	}
+
+	// PS2 (Refs #485): wasm stage scopes cargo build to the project package
+	// AND restricts to lib targets. Without `--lib`, cargo also tries to
+	// compile the project's bin (`main.rs`) for wasm32 and fails because
+	// server-only deps (tokio with full features) are not wasm-compatible.
+	#[rstest]
+	fn wasm_stage_scopes_cargo_build_to_lib_target(mut minimal_signals: DockerfileSignals) {
+		// Arrange
+		minimal_signals.pages = true;
+		minimal_signals.wasm_bindgen_version = Some("0.2.100".to_string());
+
+		// Act
+		let stage = build_wasm_stage(&minimal_signals);
+
+		// Assert
+		assert!(
+			stage_contains_run(
+				&stage,
+				"cargo build --release --target wasm32-unknown-unknown --lib -p my-app",
+			),
+			"wasm stage must pass `--lib -p {{app_name}}` to cargo build; \
+			 see kent8192/reinhardt-cloud#485"
 		);
 	}
 

